### PR TITLE
GEODE-6483: Separate concerns of render.py.

### PR DESCRIPTION
**Pull request by PurelyApplied**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [n/a] Does `gradlew build` run cleanly?

- [n/a] Have you written or updated unit tests to verify your changes?

- [n/a] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Comment by PurelyApplied** [comment]
The script is now much more useful for local testing, outside the context of Concourse.  Multiple variables files can be specified, as demonstrated in the meta pipeline scripts.  Raw, command-line variables can also be specified.  These are parsed by the `yaml.load` parser, and as such accepts any valid YML or JSON string.

The script is also hardened against interpolating variables that are not specified, as the Jinja default is to interpolate unspecified variables to an empty string.  This initially caused a few errors, as things were interpolating to empty, perhaps without our collective knowledge.

For this reason, there are two `TODO`s in this PR.  Notably:

* What should the default value of  `benchmarks.baseline_branch` be?  How is it handled when this had been blank?  @balesh2 @pivotal-jbarrett, do either of you have context on that?

* What should the default value of `test.PARALLEL_GRADLE` be?  Should we permit a default value, or require that a valid test spec has this value?  I see that there is a safeguard check in the `execute_*` scripts that sets it to true.  It seems to me that we should set that here rather than interpret downstream that blank means "yes do parallel."  Thoughts, @smgoller @rhoughton-pivot?

File: ci/pipelines/geode-build/jinja.template.yml [hunk 1]
```diff
@@ -78,7 +78,10 @@ GRADLE_TASK_OPTIONS: {{ test.GRADLE_TASK_OPTIONS }}
 DUNIT_PARALLEL_FORKS: {{ test.DUNIT_PARALLEL_FORKS }}
 MAINTENANCE_VERSION: ((geode-build-branch ))
 PARALLEL_DUNIT: {{ test.PARALLEL_DUNIT }}
-PARALLEL_GRADLE: {{ test.PARALLEL_GRADLE }}
+{% if test.PARALLEL_GRADLE is defined %}
+# TODO: DO NOT COMMIT THIS CHANGE WITHOUT USING A DEFAULTING TEMPLATE.  Should this default true or false?
```
Comment (rhoughton-pivot) [inline ci/pipelines/geode-build/jinja.template.yml]: I want parallel building in Gradle to be the default.

File: ci/pipelines/pull-request/jinja.template.yml [hunk 2]
```diff
@@ -264,7 +264,10 @@ jobs:
           GRADLE_TASK_OPTIONS: ""
           MAINTENANCE_VERSION: {{repository.branch}}
           PARALLEL_DUNIT: {{test.PARALLEL_DUNIT}}
+        {% if test.PARALLEL_GRADLE is defined %}
+          # TODO: DO NOT COMMIT THIS CHANGE WITHOUT USING A DEFAULTING TEMPLATE.  Should this default true or false?
```
Comment (rhoughton-pivot) [inline ci/pipelines/pull-request/jinja.template.yml]: As before, I want parallel to be the default and tested via the pipeline, as the source of truth. Make this happen, and remove the TODO

**Comment by PurelyApplied** [comment]
Tested locally in a private concourse, but the `meta` pipeline successfully spins up and deploys the other pipes, so each `deploy_*` script seems to be working fine.

**Pull request merged**